app/src/main/java/com/example/nbtk/slider/SliderLayoutManager.kt

```diff
@@ -63,31 +92,58 @@ class SliderLayoutManager(context: Context?) : LinearLayoutManager(context) {
-            var minDistance = recyclerView.width
-            var position = -1
-            for (i in 0 until recyclerView.childCount) {
-                val child = recyclerView.getChildAt(i)
-                val childCenterX = getDecoratedLeft(child) + (getDecoratedRight(child) - getDecoratedLeft(child)) / 2
-                var newDistance = Math.abs(childCenterX - recyclerViewCenterX)
-                if (newDistance < minDistance) {
-                    minDistance = newDistance
-                    position = recyclerView.getChildLayoutPosition(child)
-                }
+        if (state != RecyclerView.SCROLL_STATE_IDLE) return
+
+        if (orientation == HORIZONTAL) setOnScrollStateChangedHorizontal()
+        else setOnScrollStateChangedVertical()
+    }
+
+    private fun setOnScrollStateChangedHorizontal() {
+        // Find the closest child to the recyclerView center --> this is the selected item.
+        val recyclerViewCenterX = getRecyclerViewCenterX()
+        var minDistance = recyclerView.width
+        var position = -1
+        for (i in 0 until recyclerView.childCount) {
+            val child = recyclerView.getChildAt(i)
+            val childCenterX =
+                getDecoratedLeft(child) + (getDecoratedRight(child) - getDecoratedLeft(child)) / 2
+            val newDistance = Math.abs(childCenterX - recyclerViewCenterX)
+            if (newDistance < minDistance) {
+                minDistance = newDistance
+                position = recyclerView.getChildLayoutPosition(child)
             }
+        }
+
+        // Notify on item selection
+        callback?.onItemSelected(position)
+    }
 
-            // Notify on item selection
-            callback?.onItemSelected(position)
+    private fun setOnScrollStateChangedVertical() {
+        // Find the closest child to the recyclerView center --> this is the selected item.
+        val recyclerViewCenterY = getRecyclerViewCenterY()
+        var minDistance = recyclerView.height
+        var position = -1
+        for (i in 0 until recyclerView.childCount) {
+            val child = recyclerView.getChildAt(i)
+            val childCenterY =
+                getDecoratedTop(child) + (getDecoratedBottom(child) - getDecoratedTop(child)) / 2
+            val newDistance = Math.abs(childCenterY - recyclerViewCenterY)
+            if (newDistance < minDistance) {
+                minDistance = newDistance
+                position = recyclerView.getChildLayoutPosition(child)
+            }
         }
+
+        // Notify on item selection
+        callback?.onItemSelected(position)
     }
 
-    private fun getRecyclerViewCenterX() : Int {
-        return (recyclerView.right - recyclerView.left)/2 + recyclerView.left
+    private fun getRecyclerViewCenterX(): Int {
+        return (recyclerView.right - recyclerView.left) / 2 + recyclerView.left
     }
 
+    private fun getRecyclerViewCenterY() = (recyclerView.bottom - recyclerView.top) / 2 + recyclerView.top
```

Comment:
just 
private fun getRecyclerViewCenterY() = (recyclerView.bottom - recyclerView.top) / 2 
is right without extra adding recyclerView.top